### DISCUSSION TITLE
issues with building from github / modifying Valhalla copy

### DISCUSSION THREAD

**Discussion by baileyheading** [gis-ops/pyvalhalla#17] [General]
Hi, big fan of this project, very useful.

The Pyvalhalla pip package works great, but I need to make some modifications to my copy of Valhalla.

I'm not sure if I'm missing some extra steps for building on windows here.

When I try to build, setup.py doesn't find Conan.

I've tried manually calling "Conan install conanfile.txt" which allowed me to get past the errors, but I then can't get what appears in build to work with python anyway.

Is it possible to modify Valhalla from the Pip version or is a rebuild required each time Valhalla is changed?

Thanks,

**Comment by nilsnolde** [comment]
jep, you'll need to re-build the whole package every time you change within valhalla core. honestly, the build pipeline for this project is horrible, but it's a necessary ugliness to avoid millions of ci hours on every push. it was an unpleasant welcome to platform-independent packaging of c++ code. if you just need to build the python bindings for your platform and python version, you surely better build them from https://github.com/valhalla/valhalla, far less complex. if you do need to build for other platforms & python versions and need support doing so, you can always ask us for commercial support at enquiry@gis-ops.com.
